src/App.tsx

```diff
@@ -1,60 +1,125 @@
-import React from 'react';
+/* eslint-disable object-curly-newline */
+import React, { useCallback, useEffect, useMemo, useState } from 'react';
 import 'bulma/bulma.sass';
 import '@fortawesome/fontawesome-free/css/all.css';
 import './App.scss';
 
-import classNames from 'classnames';
+import cn from 'classnames';
+
+import { User } from './types/User';
+import { Post } from './types/Post';
+
+import { getUsers } from './api/users';
+import { getUserPosts } from './api/posts';
+
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
 
 export const App: React.FC = () => {
+  const [users, setUsers] = useState<User[]>([]);
+  const [userPosts, setUserPosts] = useState<Post[]>([]);
+
+  const [activeUser, setActiveUser] = useState<User | null>(null);
+  const [activePost, setActivePost] = useState<Post | null>(null);
+
+  const [isLoaderActive, setIsLoaderActive] = useState(false);
+  const [isFormActive, setIsFormActive] = useState(false);
+  const [loadPostsError, setLoadPostsError] = useState(false);
+
+  useEffect(() => {
+    getUsers()
+      .then(setUsers)
+      .catch(() => {})
+      .finally(() => {});
```

Comment:
Why do you need these blocks?
If you didn't use them it's better to remove them.